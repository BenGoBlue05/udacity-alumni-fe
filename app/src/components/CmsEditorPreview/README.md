## CmsEditorPreview Component
A component that ...

### Example

```js
<CmsEditorPreview />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
