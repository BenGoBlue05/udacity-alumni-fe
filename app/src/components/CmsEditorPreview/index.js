import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Layer from 'grommet-udacity/components/Layer';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import Markdown from 'grommet-udacity/components/Markdown';

const CmsEditorPreview = ({
  title,
  content,
  isShowing,
  onClose,
}) => (
  <Layer
    onClose={onClose}
    closer
    align="center"
    hidden={!isShowing}
    className={styles.cmsEditorPreview}
  >
    <Box
      size="large"
      pad="large"
    >
      <Heading align="center">
        {title ? title : 'No Title Set'}
      </Heading>
      {content ?
        <Markdown content={content} />
      :
        <Heading align="center" tag="h5">
          No Content
        </Heading>
      }
    </Box>
  </Layer>
);

CmsEditorPreview.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  isShowing: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default cssModules(CmsEditorPreview, styles);
