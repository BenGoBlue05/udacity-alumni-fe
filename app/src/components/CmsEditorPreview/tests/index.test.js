import CmsEditorPreview from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<CmsEditorPreview />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <CmsEditorPreview />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
