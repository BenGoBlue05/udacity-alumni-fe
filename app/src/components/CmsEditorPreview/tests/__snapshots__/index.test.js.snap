exports[`<CmsEditorPreview /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={true}>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="large"
    responsive={true}
    size="large"
    tag="div">
    <Heading
      align="center"
      tag="h1">
      No Title Set
    </Heading>
    <Heading
      align="center"
      tag="h5">
      No Content
    </Heading>
  </Box>
</Layer>
`;
